### DISCUSSION TITLE
How NumRotatableBonds works?

### DISCUSSION THREAD

**Discussion by Nokimann** [rdkit/rdkit#6597] [Q&A] [answered]
```
m = Chem.MolFromSmiles('C(=O)O')
m = Chem.AddHs(m)

print(Descriptors.NumRotatableBonds(m), Lipinski.NumRotatableBonds(m), rdMolDescriptors.CalcNumRotatableBonds(m), rdMolDescriptors.CalcNumRotatableBonds(m, strict=True))
```

It prints `0 0 0 0`

It is supposed to be one rotatable bond on the sigma bond between C and O in formic acid, isn't it?

How does `NumRotatableBonds` count the number of rotatable bonds? Or, did I misunderstand the rotatable bonds?

---

Here are some results similar to `C(=O)O`
I guess the problem comes from the double bond between O and C with OH in `C(=O)O`
Is it a matter of my lack of chemistry knowledge?

OH could be rotatable. [link](https://chem.libretexts.org/Courses/Sacramento_City_College/SCC%3A_Chem_420_-_Organic_Chemistry_I/Text/02%3A_Structure_and_Properties_of_Organic_Molecules/2.06%3A_Bond_Rotation)

(The number of rotatable bonds calculated from RDkit)

```
m = Chem.MolFromSmiles('C(=O)[OH]')
m = Chem.AddHs(m)
```

Number of rotatable bonds: 0

```
m = Chem.MolFromSmiles('CC(=O)O')
m = Chem.AddHs(m)
```

Number of rotatable bonds: 1

```
m = Chem.MolFromSmiles('C(=O)O[O]')
m = Chem.AddHs(m)
```

Number of rotatable bonds: 0

```
m = Chem.MolFromSmiles('C(=O)C')
m = Chem.AddHs(m)
```

Number of rotatable bonds: 1

```
m = Chem.MolFromSmiles('CO')
m = Chem.AddHs(m)
```

Number of rotatable bonds: 1

Similar, but this has one rotatable bond.

```
m = Chem.MolFromSmiles('C(=O)[CH]')
m = Chem.AddHs(m)
```

Number of rotatable bonds: 1

But, this has 0 rotatable bond.

```
m = Chem.MolFromSmiles('C(=O)[NH]')
m = Chem.AddHs(m)
```

Number of rotatable bonds: 0

---

I found that

```
m = Chem.MolFromSmiles('C(=O)O')
m = Chem.AddHs(m)

print(rdMolDescriptors.CalcNumRotatableBonds(m, strict=0))
```

gives a result: 1.

What does `strict` actually do? Hard to find the detailed information about `strict`.

**Comment by DavidACosgrove** [comment]
The rotatable bonds don't include terminal atoms.

**Reply by Nokimann** [reply]
Thank you for the reply. However, the below prints 2.

```
m = Chem.MolFromSmiles('CCC')
m = Chem.AddHs(m)

print(rdMolDescriptors.CalcNumRotatableBonds(m, strict=True))
```

What is the definition of terminal atoms?

**Comment by DavidACosgrove** [comment]
Yep, that's a bit odd.  I've had a look at the underlying code and I can't understand it either.

**Comment by ptosco** [comment, marked as answer]
@Nokimann The relevant bit of code is here:
https://github.com/rdkit/rdkit/blob/de602c88809ea6ceba1e8ed50fd543b6e406e9c4/Code/GraphMol/Descriptors/Lipinski.cpp#L108
If you choose `strict` definitions, rotatable bonds are identified through a more restrictive SMARTS expression that excludes, among others, amide/imide/ester/acid bonds since they have a partial double bond character,
If you choose non-`strict`, then the SMARTS is much more lenient and will consider amide/imide/ester/acid bonds as rotatable.
